core/station/impl/src/migration.rs

```diff
@@ -135,16 +149,503 @@ fn apply_migration() {
+            pub standard: Option<BlockchainStandard>,
+
+            // new fields
+            pub assets: Option<Vec<AssetId>>,
+        }
+
+        let pre_migration_entry = PreMigrationAddAccountOperationInput::deserialize(deserializer)?;
+
+        #[cfg(test)]
+        if pre_migration_entry.assets.is_none() {
+            MIGRATED_ENTRIES.with(|entries| {
+                *entries.borrow_mut() += 1;
+            });
+        }
+
+        Ok(AddAccountOperationInput {
+            name: pre_migration_entry.name,
+            metadata: pre_migration_entry.metadata,
+            read_permission: pre_migration_entry.read_permission,
+            configs_permission: pre_migration_entry.configs_permission,
+            transfer_permission: pre_migration_entry.transfer_permission,
+            configs_request_policy: pre_migration_entry.configs_request_policy,
+            transfer_request_policy: pre_migration_entry.transfer_request_policy,
+            assets: pre_migration_entry
+                .assets
+                .unwrap_or_else(|| vec![INITIAL_ICP_ASSET_ID]),
+        })
+    }
+}
+
+impl<'de> Deserialize<'de> for AddAddressBookEntryOperationInput {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        #[allow(dead_code)]
+        #[derive(Debug, Deserialize)]
+        struct PreMigrationAddAddressBookEntryOperationInput {
+            pub address_owner: String,
+            pub address: String,
+            pub blockchain: Blockchain,
+            #[serde(default)]
+            pub labels: Vec<String>,
+            pub metadata: Vec<MetadataItem>,
+
+            // added fields
+            pub address_format: Option<AddressFormat>,
+        }
+
+        let pre_migration_entry =
+            PreMigrationAddAddressBookEntryOperationInput::deserialize(deserializer)?;
+
+        #[cfg(test)]
+        if pre_migration_entry.address_format.is_none() {
+            MIGRATED_ENTRIES.with(|entries| {
+                *entries.borrow_mut() += 1;
+            });
+        }
+
+        Ok(AddAddressBookEntryOperationInput {
+            address_owner: pre_migration_entry.address_owner,
+            address: pre_migration_entry.address,
+            blockchain: pre_migration_entry.blockchain,
+            labels: pre_migration_entry.labels,
+            metadata: pre_migration_entry.metadata,
+            address_format: pre_migration_entry
+                .address_format
+                .unwrap_or(AddressFormat::ICPAccountIdentifier),
+        })
+    }
+}
+
+impl RebuildRepository<AddressBookEntryKey, AddressBookEntry, VirtualMemory<Memory>>
+    for AddressBookRepository
+{
+}
+
+impl RebuildRepository<TransferKey, Transfer, VirtualMemory<Memory>> for TransferRepository {}
+impl RebuildRepository<AccountKey, Account, VirtualMemory<Memory>> for AccountRepository {}
+impl RebuildRepository<RequestKey, Request, VirtualMemory<Memory>> for RequestRepository {}
```

Comment:
we don't seem to be calling `rebuild` for these repositories inside `apply_migration` to rebuild them

Reply:
Ah good catch!